bentoml/_internal/cli/click_utils.py

```diff
@@ -180,19 +195,27 @@ def wrapper(func):
 
         return wrapper
 
+    def resolve_command(
+        self, ctx: click.Context, args: t.List[str]
+    ) -> t.Tuple[str, click.Command, t.List[str]]:
+        try:
+            return super(BentoMLCommandGroup, self).resolve_command(ctx, args)
+        except UsageError as e:
+            error_msg = str(e)
+            original_cmd_name = click.utils.make_str(args[0])
+            matches = difflib.get_close_matches(
+                original_cmd_name, self.list_commands(ctx), 3, 0.5
+            )
+            if matches:
+                fmt_matches = "\n    ".join(matches)
+                error_msg += "\n\n"
+                error_msg += f"Did you mean?\n    {fmt_matches}"
+            raise UsageError(error_msg, e.ctx)
```

Comment:
@aarnphm this is really nice! 

@yubozhao @jjmachan let's do something similar in bentoctl?